test/mx/test_transform_equals.py

```diff
@@ -0,0 +1,219 @@
+        instance_sampler=transform.ContinuousTimePointSampler(),
+    )
+
+    splitter2 = transform.ContinuousTimeInstanceSplitter(
+        past_interval_length=1,
+        future_interval_length=1,
+        instance_sampler=transform.ContinuousTimePointSampler(),
+    )
+
+    splitter3 = transform.ContinuousTimeInstanceSplitter(
+        past_interval_length=1,
+        future_interval_length=1,
+        instance_sampler=transform.ContinuousTimePointSampler(min_past=1.0),
+    )
+    assert equals(splitter, splitter2)
+    assert not equals(splitter, splitter3)
+
+
+def test_batch():
+    assert equals(Batch(batch_size=10), Batch(batch_size=10))
+    assert not equals(Batch(batch_size=10), Batch(batch_size=100))
+
+
+def test_chain():
+
+    chain = transform.Chain(
+        trans=[
+            transform.AddTimeFeatures(
+                start_field=FieldName.START,
+                target_field=FieldName.TARGET,
+                output_field="time_feat",
+                time_features=[
+                    time_feature.DayOfWeek(),
+                    time_feature.DayOfMonth(),
+                    time_feature.MonthOfYear(),
+                ],
+                pred_length=10,
+            ),
+            transform.AddAgeFeature(
+                target_field=FieldName.TARGET,
+                output_field="age",
+                pred_length=10,
+                log_scale=True,
+            ),
+            transform.AddObservedValuesIndicator(
+                target_field=FieldName.TARGET, output_field="observed_values"
+            ),
+        ]
+    )
+
+    assert equals(chain, serde.flat.clone(chain))
+    assert not equals(chain, serde.flat.clone(chain, {"trans": []}))
+
+    another_chain = transform.Chain(
+        trans=[
+            transform.AddTimeFeatures(
+                start_field=FieldName.START,
+                target_field=FieldName.TARGET,
+                output_field="time_feat",
+                time_features=[
+                    time_feature.DayOfWeek(),
+                    time_feature.DayOfMonth(),
+                    time_feature.MonthOfYear(),
+                ],
+                pred_length=10,
+            ),
+            transform.AddAgeFeature(
+                target_field=FieldName.TARGET,
+                output_field="age",
+                pred_length=10,
+                log_scale=False,
+            ),
+            transform.AddObservedValuesIndicator(
+                target_field=FieldName.TARGET, output_field="observed_values"
+            ),
+        ]
+    )
+    assert not equals(chain, another_chain)
+
+
+def test_gluon_predictor():
+    train_length = 100
+    pred_length = 10
+
+    estimator = CanonicalRNNEstimator("5min", train_length, pred_length)
+
+    assert equals(estimator, serde.flat.clone(estimator))
+    assert not equals(estimator, serde.flat.clone(estimator, {"freq": "1h"}))
+
+
+if __name__ == "__main__":
+    test_filter_transformation()
```

Comment:
Let's remove this.

There is the `-k` flag in pytest, where you can easily select tests, so you can do `pytest test/mx/test_transform_equals.py -k filter` to just run tests containing the word `filter`.

Reply:
Removed. I think one can also use `pytest test/mx/test_transform_equals.py::test_filter_transformation`.

Reply:
True, but I'm also lazy :)